packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemFileHandle.kt

```diff
@@ -0,0 +1,49 @@
+package expo.modules.filesystem.next
+
+import expo.modules.kotlin.sharedobjects.SharedObject
+import java.nio.ByteBuffer
+import java.nio.channels.FileChannel
+class FileSystemFileHandle(file: FileSystemFile) : SharedObject() {
+  val fileChannel: FileChannel
+
+  init {
+    fileChannel = file.file.inputStream().channel
+  }
+  fun close() {
```

Comment:
It isn't enough. You should also override the `deallocate` function to close the stream when the JS object is deallocated.